core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java

```diff
@@ -461,8 +463,9 @@ private OAuthTokenResponse refreshCurrentToken(RESTClient client) {
       }
     }
 
-    static boolean isExpired(Long expiresAtMillis, long now) {
-      return expiresAtMillis != null && expiresAtMillis < now;
+    static boolean isExpired(Instant expiresAt) {
```

Comment:
@rdblue you mentioned in https://github.com/apache/iceberg/pull/6489#discussion_r1072451113 that this type of method can be misleading, because the token could still be expired even though `expiresAt` is `null`.
To me it makes sense to have this method, given that we're using it at 2 different places. 
Technically we're just checking whether a given `Instant` is expired or not so IMO it's clear what the purpose of this method is. 
I can't come up with a better name, but I'm happy to rename it to something more obvious in case you have suggestions. 
Would `isNonNullAndExpired()` be more descriptive about the purpose of the method?

Reply:
I don't think we need a new method just because we do a null check twice. The null check and comparison is clear and readable.

Reply:
it is less about the null check and more about the expiration check that we repeat at 2 places

Reply:
I've removed the method and reverted the code to how it was before